### PR TITLE
GCB-349: Wrap widget with header view.

### PR DESCRIPTION
**Pull request by krris**
Make sure these boxes are checked before submitting your Pull Request - thank you!

- [x] [SwiftLint](https://github.com/Realm/SwiftLint) is installed on your system
- [x] The project does not generate any warnings or errors
- [x] The tests are passing
- [x] Branch name is in line with [GitFlow](http://jeffkreeftmeijer.com/2010/why-arent-you-using-git-flow/)

### CODE REVIEW

File: GrandCentralBoard/Widgets/Image/ImageWidget.swift [hunk 1]
```diff
@@ -20,7 +20,11 @@ final class ImageWidget: WidgetControlling {
     }
 
     var view: UIView {
-        return widgetView
+        guard isHeaderVisible else { return widgetView }
+
+        let viewModel = WidgetTemplateViewModel(title: "CAT PHOTOS", subtitle: "NEWEST CAT PROFILES", contentView: widgetView)
+        let layoutSettings = WidgetTemplateLayoutSettings(contentMargin: UIEdgeInsetsZero, displayContentUnderHeader: true)
+        return WidgetTemplateView.viewWithViewModel(viewModel, layoutSettings: layoutSettings)
```
Comment (ncreated) [inline GrandCentralBoard/Widgets/Image/ImageWidget.swift]: Similar issue to https://github.com/macoscope/GrandCentralBoard/pull/140#discussion_r63876301. Let's construct this wrapper view somewhere else.

**Comment by ncreated** [comment]
@krris 🎾

File: GrandCentralBoard/Widgets/Image/ImageWidget.swift [hunk 2]
```diff
@@ -9,18 +9,25 @@ import GCBCore
...
     let isHeaderVisible: Bool
 
     init(view: ImageWidgetView, sources: [UpdatingSource], isHeaderVisible: Bool) {
         self.widgetView = view
         self.sources = sources
         self.isHeaderVisible = isHeaderVisible
+
+        let viewModel = WidgetTemplateViewModel(title: "CAT PHOTOS", subtitle: "NEWEST CAT PROFILES", contentView: widgetView)
+        let layoutSettings = WidgetTemplateLayoutSettings(contentMargin: UIEdgeInsetsZero, displayContentUnderHeader: true)
+        widgetViewWrapper = WidgetTemplateView.viewWithViewModel(viewModel, layoutSettings: layoutSettings)
```
Comment (ncreated) [inline GrandCentralBoard/Widgets/Image/ImageWidget.swift]: Sorry, I didn't see it yesterday in CR. If `isHeaderVisible` equals `false` we create `WidgetTemplateView`, but never use it. If we know that we don't need a header, let's not create it. What about:

```swift
private let widgetView: ImageWidgetView
private let mainView: UIView

// ...

if (isHeaderVisible) {
   let wrapper = ...
   widgetView = view
   mainView = wrapper
} else {
   widgetView = view
   mainView = view
}

// ...

var view: UIView {
   return mainView
}

```
?

**Comment by ncreated** [comment]
@krris ⚽

File: GrandCentralBoard/Widgets/Image/ImageWidget.swift (hunk 2 above)
Comment (krris) [inline GrandCentralBoard/Widgets/Image/ImageWidget.swift]: Good point 👍

**Comment by krris** [comment]
@ncreated 🎾

**Pull request merged**